### PR TITLE
Update publishing-docker-images.md

### PR DESCRIPTION
**Pull request by jfoclpf**
outdated actions, see https://docs.docker.com/build/ci/github-actions/

### Why:

Closes: 

### Check off the following:

- [x] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline.

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [ ] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

**Comment by cmwilson21** [comment]
@jfoclpf Thanks so much for opening a PR! I'll get this triaged for review :zap:

File: content/actions/publishing-packages/publishing-docker-images.md [hunk 1]
```diff
@@ -93,7 +93,7 @@ jobs:
         uses: {% data reusables.actions.action-checkout %}
       
       - name: Log in to Docker Hub
-        uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
+        uses: docker/login-action@v2
```
Comment (skedwards88) [inline content/actions/publishing-packages/publishing-docker-images.md:96]:
```suggestion
        uses: docker/login-action@f4ef78c080cd8ba55a85445d5b36e214a81df20a
```

File: content/actions/publishing-packages/publishing-docker-images.md [hunk 2]
```diff
@@ -174,14 +175,14 @@ jobs:
         uses: {% data reusables.actions.action-checkout %}
       
       - name: Log in to GitHub Docker Registry
-        uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
+        uses: docker/login-action@v2
```
Comment (skedwards88) [inline content/actions/publishing-packages/publishing-docker-images.md:178]:
```suggestion
        uses: docker/login-action@f4ef78c080cd8ba55a85445d5b36e214a81df20a
```

File: content/actions/publishing-packages/publishing-docker-images.md [hunk 3]
```diff
@@ -226,7 +227,7 @@ jobs:
         uses: {% data reusables.actions.action-checkout %}
       
       - name: Log in to Docker Hub
-        uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
+        uses: docker/login-action@v2
```
Comment (skedwards88) [inline content/actions/publishing-packages/publishing-docker-images.md:230]:
```suggestion
        uses: docker/login-action@f4ef78c080cd8ba55a85445d5b36e214a81df20a
```

File: content/actions/publishing-packages/publishing-docker-images.md [hunk 4]
```diff
@@ -174,14 +175,14 @@ jobs:
...
         with:
           registry: {% ifversion ghae %}docker.YOUR-HOSTNAME.com{% else %}docker.pkg.github.com{% endif %}
           username: {% raw %}${{ github.actor }}{% endraw %}
           password: {% raw %}${{ secrets.GITHUB_TOKEN }}{% endraw %}
       
       - name: Build and push Docker image
-        uses: docker/build-push-action@f2a1d5e99d037542a71f64918e516c093c6f3fc4
+        uses: docker/build-push-action@v4
```
Comment (skedwards88) [inline content/actions/publishing-packages/publishing-docker-images.md:185]:
```suggestion
        uses: docker/build-push-action@3b5e8027fcad23fda98b2e3ac259d8d67585f671
```

File: content/actions/publishing-packages/publishing-docker-images.md [hunk 5]
```diff
@@ -247,7 +248,7 @@ jobs:
             {% ifversion fpt or ghec or ghes > 3.4 %}{% data reusables.package_registry.container-registry-hostname %}/{% raw %}${{ github.repository }}{% endraw %}{% elsif ghae %}{% raw %}docker.YOUR-HOSTNAME.com/${{ github.repository }}/my-image{% endraw %}{% else %}{% raw %}docker.pkg.github.com/${{ github.repository }}/my-image{% endraw %}{% endif %}
       
       - name: Build and push Docker images
-        uses: docker/build-push-action@f2a1d5e99d037542a71f64918e516c093c6f3fc4
+        uses: docker/build-push-action@v4
```
Comment (skedwards88) [inline content/actions/publishing-packages/publishing-docker-images.md:251]:
```suggestion
        uses: docker/build-push-action@3b5e8027fcad23fda98b2e3ac259d8d67585f671
```

File: content/actions/publishing-packages/publishing-docker-images.md [hunk 6]
```diff
@@ -105,9 +105,10 @@ jobs:
           images: my-docker-hub-namespace/my-docker-hub-repository
       
       - name: Build and push Docker image
-        uses: docker/build-push-action@f2a1d5e99d037542a71f64918e516c093c6f3fc4
+        uses: docker/build-push-action@v4
```
Comment (skedwards88) [inline content/actions/publishing-packages/publishing-docker-images.md:108]:
```suggestion
        uses: docker/build-push-action@3b5e8027fcad23fda98b2e3ac259d8d67585f671
```

**Review by skedwards88 — COMMENTED** [review]
Thanks for updating this!

We recommend that you pin third party actions to a SHA instead of to a version: https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions Unless the action is owned by GitHub, we aim to follow this practice in all of our examples.

I left suggestions to use the SHA corresponding to the latest release of each action. I'll incorporate those changes then get this merged down for you.

**Review by skedwards88 — APPROVED** [review]

**Pull request merged**